packages/arb-token-bridge-ui/src/util/networks.ts

```diff
@@ -307,87 +306,41 @@ export type SwitchChainProps = {
   onSwitchChainNotSupported?: (attemptedChainId: number) => void
 }
 
-const isSwitchChainSupported = (provider: ExtendedWeb3Provider) => {
-  const { provider: innerProvider } = provider
-  return innerProvider.isMetaMask || innerProvider.isImToken
-}
-
-// eslint-disable-next-line @typescript-eslint/no-empty-function
-const noop = () => {}
-
-const onSwitchChainNotSupportedDefault = (attemptedChainId: number) => {
+const onSwitchChainNotSupported = (attemptedChainId: number, isTx: boolean) => {
   const isDeposit = isNetwork(attemptedChainId).isEthereum
   const targetTxName = isDeposit ? 'deposit' : 'withdraw'
   const networkName = getNetworkName(attemptedChainId)
 
+  const message = isTx
+    ? `Please connect to ${networkName} on your wallet before signing your ${targetTxName} transaction.`
+    : `Please connect to ${networkName} on your wallet.`
+
   // TODO: show user a nice dialogue box instead of
   // eslint-disable-next-line no-alert
-  alert(
-    `Please connect to ${networkName} to ${targetTxName}; make sure your wallet is connected to ${networkName} when you are signing your ${targetTxName} transaction.`
-  )
+  alert(message)
 }
 
-export async function switchChain({
-  chainId,
-  provider,
-  onSuccess = noop,
-  onError = noop,
-  onSwitchChainNotSupported = onSwitchChainNotSupportedDefault
-}: SwitchChainProps) {
-  // do an early return if switching-chains is not supported by provider
-  if (!isSwitchChainSupported(provider)) {
-    onSwitchChainNotSupported?.(chainId)
+export function handleSwitchNetworkError(
+  error: any,
+  { chainId }: SwitchNetworkArgs,
+  context: unknown = { isTx: false }
```

Comment:
We don't seem to be using `isTx` in any of it's function calls. Or maybe I am not understanding it right. Can you pls explain this?

Reply:
when it's not explicitly passed in, it's using the default value `false` here